config/sample-idex.cfg

```diff
@@ -90,7 +114,149 @@ gcode:
+        # Update core Klipper's fan speed to the fan speed of the active toolhead
+        # Only do this if you have a sacrificial [fan] section
+        M106.1 S{printer["fan_generic fan_extruder1"].speed * 255}
+    {% endif %}
     ACTIVATE_EXTRUDER EXTRUDER=extruder1
     SET_DUAL_CARRIAGE CARRIAGE=1
-    SET_GCODE_OFFSET Y=15
+    {% if printer["gcode_macro T0"].offset_applied == 0 %}
+        SET_GCODE_OFFSET  X_ADJUST={ svv.xoffset } Y_ADJUST={ svv.yoffset }
+        SET_GCODE_OFFSET Z_ADJUST={ svv.zoffset } MOVE=1
+        SET_GCODE_VARIABLE MACRO=T0 VARIABLE=offset_applied VALUE=1
+    {% endif %}
+#    SET_INPUT_SHAPER if nessesary, update input shaping for the second carraige
+
+# YOU MUST CALIBRATE YOUR ROTATION_DISTANCE BEFORE THIS IS USEFUL!!!
+# Configure this to draw 2 line segments perpendicular to the dual_carraige axis
+# that meet in the middle of the bed. Both lines should have the same coordinate
+# on the axis of the dual_carraige. If they don't print inline, adjust your endstop
+# settings.
+[gcode_macro set_separation]
+gcode:
+    {% set svv = printer.save_variables.variables %}
+
+    {% set oldX = svv.xoffset|float %}
+    {% set oldY = svv.yoffset|float %}
+    {% set oldZ = svv.zoffset|float %}
+
+    {% if params.X is defined %}
+        SAVE_VARIABLE VARIABLE=xoffset VALUE={ params.X|float }
+    {% endif %}
+
+    {% if params.Y is defined %}
+        SAVE_VARIABLE VARIABLE=yoffset VALUE={ params.Y|float }
+    {% endif %}
+
+    {% if params.Z is defined %}
+        SAVE_VARIABLE VARIABLE=zoffset VALUE={ params.Z|float }
+    {% endif %}
+
+    {% if params.X_ADJUST is defined %}
+        {% set newX = params.X_ADJUST|float + oldX %}
+        SAVE_VARIABLE VARIABLE=xoffset VALUE={ newX|float }
+    {% endif %}
+
+    {% if params.Y_ADJUST is defined %}
+        {% set newY = params.Y_ADJUST|float + oldY %}
+        SAVE_VARIABLE VARIABLE=yoffset VALUE={ newY|float }
+    {% endif %}
+
+    {% if params.Z_ADJUST is defined %}
+        {% set newZ = params.Z_ADJUST|float + oldZ %}
+        SAVE_VARIABLE VARIABLE=yoffset VALUE={ newZ|float }
+    {% endif %}
+
+[gcode_macro calibrate_separation]
+gcode:
+    G28
+    G90
+    M83
+    T0 ; test T0
+    G1 X120 Y150 Z.2 F4800
+    G1 Y75 E10
+    T1 ; test T1
+    G1 X120 Y0 Z.2
+    G1 Y75 E10
+
+# The following is for if your carraiges have their own part cooling fans.
+# Since this macro overrides M106, you cannot also have a [fan] section defined.
+# If your printer only has one part cooling fan, don't copy this section
+[fan_generic part_fan_0]
+pin: PE5
+
+[fan_generic part_fan_1]
+pin: PD13
+
+# For completeness, you can add a [fan] section with an unused pin
+[fan]
+pin: rpi:gpio20
+
+[gcode_macro M106]
+# Only rename_existing if you have a sacrificial [fan] section
+rename_existing: M106.1
+# The variable that controls fan speed swopping if not specifying P parameter
+# -1 means the control is disabled, a value of 0-1 is the requested fan speed.
```

Comment:
It could be just me, I got confused a bit at first by a description of P parameter. Maybe you could say that P is the index of a fan (0-1)? *And -1 is the active extruder fan.